### DISCUSSION TITLE
Organ ready indicator light

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#1114] [General]
Hello all, for an instrument that does not have screens (touchscreen or not), but only tie rods or stop register dominoes, is there in GrandOrgue a way to have a warning light or other alert indicating when is the organ operational?

Thank you :) !

**Comment by eturpault** [comment]
Hello @Jean-Gabriel1982, I think that the easiest way to know when the organ is operational, without visible feedback, is to have a blower noise or ambiant noise or single sound which is played by default at organ start (i.e. defined in an invisible stop engaged by default). So when you hear this sound you know that the organ is up and running.

**Reply by Jean-Gabriel1982** [reply]
Hello @eturpault thank you very much, yes that is an option, but in a church we have to listen carefully for the sound ... and if there is some noise around it is difficult...
The way to increase the level of this sound is an option too ..

I thought on a kind of green light when the organ is ready ..

**Reply by eturpault** [reply]
Such a green light should have to be powered by a physical interface which the on/off state is set by GO. According to what I know the only physical interfaces on which GO can write are screen, MIDI and Audio devices. So if the screen and audio solutions doesn't work for you, an alternative solution could be to emit on the MIDI output of the computer a MIDI event which is associated to the blower switch status, and on the organ physical console to have a light which the state can be changed by an external MIDI event.

**Reply by davidgritter** [reply]
you should be able to attach such a light to a stop that is saved in your configuration as on.  Set up the send on that stop to emit the midi message that will turn your light on.   I used to do this with a  headless organ simply by leaving one stop on when saving. The next time I powered up that stop would light up when the organ was ready to play

**Reply by willeke1234** [reply]
Why do you use the blower switch or a stop? Does the ON "LED" on the Master Controls panel not work?

**Reply by davidgritter** [reply]
I didn't know about the "on" button; don't use the other panels much.  Taking a look at it shows it can send a midi message, so it should work.

**Reply by William-Purvis** [reply]
I have reached the point where I need this, did a quick scan on the forum and found this.
Yes, indeed, the Control Panel 'ON' lamp is quite happy to send a MIDI message to my scanners (Arduini).
I've finally reached the stage where my console can be wired into the PC, and this was on my list of things I needed
to resolve. A quick test on my laptop with MIDI output connected to a midi monitor. Right click on the ON lamp and
fill in the details. I have 64 rocker switches for tab stops and 2 dozen pistons. I have three Arduini which do the scanning, 
and also control LEDs on the rockers and buttons. When the organ gets power from the wall, the piston scanner plays chase
with the LEDS until the PC has booted up and started GO. I needed something to tell it to stop fooling about and this is
just what I need. The piston scanner monitors the MIDI messages which go to the tab stops to set their LEDs and
by setting the tab #127 (which doesn't physically exist) I can stop flashing the LEDs so the organist will know he can
start playing. I could equally add an LED somewhere else on the console, to indicate organ ready, but this seemed
to be more fun!

**Reply by William-Purvis** [reply]
I now think the On lamp comes on too soon - it comes on as GO starts to load the organ samples, so it's not a good indication of it being ready to play. Is there any other item I can set up to send a ready signal when the samples have all been loaded?

**Reply by larspalo** [reply]
You could add a switch with default to engaged (and GCState=-1) somewhere (on any panel in the sample set you're going to use) and use it to send the MIDI message you need to indicate that the organ is ready. If the sample set already have a drawstop type that does the same thing (for instance a blower or sperrventil or anything similar that default to engaged and have GCState=-1) you could simply use that instead of creating a new one.

On the other hand, if the On button on the master panel is firing too early, then that could be considered a bug that should be fixed.

**Reply by willeke1234** [reply]
@William-Purvis On my computer the ON "LED" sends the on-event when the samples are loaded and the organ is ready. What are your MIDI event settings for the ON "LED"? Does it send a MIDI event when it turns off?

**Comment by davidgritter** [comment]
I used to set one gui  stop or switch element active on a panel as lars suggests, and when the organ was loaded the stop switch lamp for that element turned on, not before as you report the on button does.  However, if that doesn't work you could have your arduino send noteon/noteoff messages in sequence to a dummy element and have it light the console on button when it receives replies from GrandOrgue to those messages.

**Comment by William-Purvis** [comment]
On 21/12/2023 17:16, larspalo wrote:
>
> You could add a switch with default to engaged (and GCState=-1) 
> somewhere (on any panel in the sample set you're going to use) and use 
> it to send the MIDI message you need to indicate that the organ is 
> ready. If the sample set already have a drawstop type that does the 
> same thing (for instance a blower or sperrventil or anything similar 
> that default to engaged and have GCState=-1) you could simply use that 
> instead of creating a new one.
>
> On the other hand, if the On button on the master panel is firing too 
> early, then that could be considered a bug that should be fixed.
>
Thanks for this, I'll try it out with a dummy switch.

Bill

-- 
+----------------------------------------+
| Bill Purvis                            |
| email: ***@***.***                  |
+----------------------------------------+

**Comment by William-Purvis** [comment]
What is GCState? I've not seen that anywhere so far. Is this the initial state of GO?, something to do with SetterGC?

**Comment by William-Purvis** [comment]
Sorry, found it in the ODF help under DrawStop....

**Comment by William-Purvis** [comment]
I'm trying to set up a Drawstop as suggested by Lars, but it keeps reporting the entries as unused.
I've added a drawstop entry, a switch entry, and a panel element for it, and added the switch to the list of switches on the Great manual. What more do I need to add? I can't find anything that explicitly references a drawstop in the help file.

**Reply by larspalo** [reply]
Just a switch with Input function. A switch is a drawstop without any other attributes, it doesn't need to be referenced anywhere.

A stop is also a drawstop, but it has other requirements than a switch.

**Reply by William-Purvis** [reply]
extracted from ODF:

[DrawStop001]
DefaultToEngaged=1
GCState=-1
NumberOfSwitches=1
Switch001=69
Name=Dummy Ready
...
[Switch069]
Function=Input
Name=Dummy Ready
DefaultToEngaged=Y

still get unused error messages

**Reply by larspalo** [reply]
You cannot have the [DrawStop001] entry at all, it's not valid. Just remove it

**Reply by larspalo** [reply]
And add the GCState=-1 to the switch.
